### PR TITLE
TF-2989 Fix drag and drop text within composer is broken

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2989 

## Root cause

- Due disable summernote's default `drag & drop` in `html-editor-enhanced` to customize drop zone view

## Dependent 

- https://github.com/linagora/html-editor-enhanced/pull/42
- https://github.com/linagora/html-editor-enhanced/pull/43

## Resolved

https://github.com/linagora/tmail-flutter/assets/80730648/15055d87-97dc-4a27-8c77-a603cc338ebf

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -374,22 +374,42 @@ class ComposerController extends BaseController with DragDropFileMixin implement
...
+      event.preventDefault();
+
+      if (_validateFilesTransfer(event.dataTransfer.types)) {
+        mailboxDashBoardController.localFileDraggableAppState.value = DraggableAppState.inActive;
+      }
     });
   }
 
+  bool _validateFilesTransfer(List<dynamic>? types) {
+    return types?.any((type) => type == 'Files') ?? false;
```
Comment (tddang-linagora) [inline lib/features/composer/presentation/composer_controller.dart:410]:
```suggestion
    return types?.every((type) => type == 'Files') ?? false;
```

Comment (dab246) [reply lib/features/composer/presentation/composer_controller.dart:410]: No, we just need the type list to contain at least one type of `Files`, we will give priority to drag & drop files.

**Review by tddang-linagora — APPROVED** [review]

File: pubspec.yaml [hunk 2]
```diff
@@ -60,7 +60,7 @@ dependencies:
   html_editor_enhanced:
     git:
       url: https://github.com/linagora/html-editor-enhanced.git
-      ref: cnb_supported
+      ref: bugfix/fix_drag_and_drop_text
```
Comment (hoangdat) [inline pubspec.yaml:63]: please update it

Comment (dab246) [reply pubspec.yaml]: Done

**Comment by hoangdat** [comment]
- sorry my mistake, please move all to sprint 25 branch.

**Comment by dab246** [comment]
> sorry my mistake, please move all to sprint 25 branch.
> 
> * [x]  move html to sprint's branch

@hoangdat  Done. Please review PR https://github.com/linagora/html-editor-enhanced/pull/43

**Review by hoangdat — APPROVED** [review]

**Pull request merged**